onnx/numpy_helper.py

```diff
@@ -221,7 +221,7 @@ def unpack_int4(
     return res
 
 
-def evaluate_float4e2m1_from_bits(x: np.uint8) -> np.float32:
+def evaluate_float4e2m1_from_bits(x: np.ndarray[np.uint8]) -> np.ndarray[np.float32]:
```

Comment:
I would consider calling it `float4e2m1_to_float32` or something like that.

Reply:
Cool! I didn't know `np.ndarray[]` works

Reply:
Renamed to `unpacked_float4e2m1_to_float32`.

Looks like the linter doesn't like `np.ndarray[]` :( . Changed back.

Reply:
Have you tried the following?

```python
import numpy.typing as npt

npt.NDArray(np.int8)  # something like this
```

https://numpy.org/doc/stable/reference/typing.html#numpy.typing.NDArray

Reply:
Using `npt.NDArray` seems to work. Although I don't see NDArray being used previously in the code and saw this comment:
```
if sys.version_info <= (3, 10):
    #  AttributeError: module 'numpy.typing' has no attribute 'NDArray'
    backend_test.exclude("test_image_decoder_decode_")
```
Not sure if it's still an issue.